alacritty/src/display.rs

```diff
@@ -437,7 +437,14 @@ impl Display {
         mods: ModifiersState,
         search_state: &SearchState,
     ) {
-        let grid_cells: Vec<RenderableCell> = terminal.renderable_cells(config).collect();
+        // Convert search match from viewport to absolute indexing.
+        let search_active = search_state.regex().is_some();
+        let focused_match = search_state.focused_match();
+        let focused_match_start = terminal.grid().clamp_buffer_to_visible(*focused_match.start());
+        let focused_match_end = terminal.grid().clamp_buffer_to_visible(*focused_match.end());
+        let viewport_match = focused_match_start..=focused_match_end;
```

Comment:
If you have scrollback, and you've matched something on top most line(shouldn't be top most though) and then scroll down, so this focused match will go offscreen, you'll have a top left cell with active match colors.

![picture](https://user-images.githubusercontent.com/27620401/98620303-1bf18d80-2316-11eb-819d-5f97e900b82b.png)

Also, if you don't have scrollback and resize terminal, you'll match top left cell as well, however in that case it's required that you don't have scrollback at all.

![resize_top_selected](https://user-images.githubusercontent.com/27620401/98620373-42172d80-2316-11eb-9b0e-f1c72511d524.png)